services/motion/builtin/builtin.go

```diff
@@ -167,19 +166,58 @@ func (ms *builtIn) Move(
 	return true, nil
 }
 
-// MoveOnMap will move the given component
+// MoveOnMap will move the given component to the given destination on the slam map generated from a slam service specified by slamName.
+// Bases are the only component that supports this.
 func (ms *builtIn) MoveOnMap(
 	ctx context.Context,
 	componentName resource.Name,
 	destination spatialmath.Pose,
 	slamName resource.Name,
 	extra map[string]interface{},
 ) (bool, error) {
+	operation.CancelOtherWithLabel(ctx, "motion-service")
+
+	// get the SLAM Service from the slamName
 	slamService, ok := ms.slamServices[slamName]
 	if !ok {
 		return false, errors.Wrap(resource.NewNotFoundError(slamName), "motion service missing weak dependency")
 	}
-	_ = slamService
+
+	// create a KinematicBase from the componentName
+	b, ok := ms.components[componentName]
+	if !ok {
+		return false, fmt.Errorf("only Base components are supported for MoveOnMap: could not find an Base named %v", componentName)
```

Comment:
This doesn't actually do this right now, right? This won't work properly until a base matcher is written

Reply:
the second part of the error sentence should work - it should match resource names that have components in them with that specific name. Could be skipped since it's probably going to go away for some or all component types.

The check he has under this line is the actual check for a kinematic base, so I would vote for amending the error message slightly.

Reply:
The second part of the error sentence is misleading. This is currently checking if *any* component exists with the given name, not just bases.

Yes, it's technically true that if no component was found with a given name, then no Base was found with a given name, as bases are a subset of all components.

The problem here is that if a component does exist with the given name, and it is not a Base, this is the error message that should be emitted. That is not the case.

Reply:
Updating to return a dependency not found error